packages/expo/src/useOAuth.ts

```diff
@@ -0,0 +1,84 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { OAuthStrategy } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+
+export type UseOAuthFlowParams =
+  | {
+      strategy: OAuthStrategy;
+      redirectUrl?: string;
+    }
+  | OAuthStrategy;
+
+export function useOAuth(params: UseOAuthFlowParams) {
+  let strategy: OAuthStrategy;
+  let redirectUrl: string;
+
+  if (typeof params === 'string') {
+    strategy = params;
+  } else if (typeof params === 'object') {
+    strategy = params.strategy;
+    redirectUrl = params.redirectUrl || '';
+  }
+
+  const { signIn, setSession, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startOAuthFlow() {
+    if (!isSignInLoaded || !isSignUpLoaded) {
```

Comment:
This is going to throw if someone tries to call `startOAuthFlow` without checking `isLoaded` in advance with a Typerror (probably `undefined is not a function`).

Since `startOAuthFlow` returns a promise already, we can wait for clerk to load within the function like we do for `useAuth.getToken` and `useAuth.signOut`: https://github.com/clerkinc/javascript/blob/a285a4e4c38bc1ed1cca6a910d2ec6d76b911822/packages/react/src/hooks/utils.ts#L18